tests/ui/macros/stringify.rs

```diff
@@ -685,6 +690,16 @@ fn test_stmt() {
         "let _ = match void {} - 1;",
         "let _ = match void {} - 1",
     );
+    macro_rules! c2_let_expr_else_return {
+        ([ $expr:expr ], $stmt_expected:expr, $tokens_expected:expr $(,)?) => {
+            c2!(stmt, [ let _ = $expr else { return; } ], $stmt_expected, $tokens_expected);
+        };
+    }
+    c2_let_expr_else_return!(
+        [ f() ],
+        "let _ = f() else { return; };",
+        "let _ = f() else { return; }",
+    );
```

Comment:
I initially tried testing this with:

```rust
    c2_let_expr_else_return!(
        [ loop {} ],
        "let _ = (loop {}) else { return; };",
        "let _ = loop {} else { return; }",
    );
```

but the curly brace diagnostic actually looks inside of the $expr. I am not sure whether that is intentional. (In my opinion it shouldn't do that.) I added a separate test in tests/ui/unpretty instead.

```console
error: right curly brace `}` before `else` in a `let...else` statement not allowed
  --> tests/ui/macros/stringify.rs:704:17
   |
LL |         [ loop {} ],
   |                 ^
   |
help: wrap the expression in parentheses
   |
LL |         [ (loop {}) ],
   |           +       +
```